### PR TITLE
Changes to support custom node driver UI extensions

### PR DESCRIPTION
**Pull request by nwmac**
This PR makes a few small changes, to better support custom node driver UI extensions.

This PR adds a couple of improvements:

- Adds support for machine pool validation
- Adds a busy state which can be used by machine pool components to disable their controls when the UI is busy

Machine Pool UIs will work without these changes.

### CODE REVIEW

File: shell/edit/provisioning.cattle.io.cluster/MachinePool.vue [hunk 1]
```diff
@@ -237,9 +255,11 @@ export default {
       :credential-id="credentialId"
       :pool-index="idx"
       :machine-pools="machinePools"
+      :busy="busy"
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/MachinePool.vue:258]: disabled/busy should also apply to the `AdvancedSection` input fields below

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 2]
```diff
@@ -2014,6 +2042,14 @@ export default {
       this.lastDefaultPodSecurityPolicyTemplateName = value;
     },
 
+    // Track Machine Pool validation status
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:2045]: jsdoc

**Review by richard-cox — CHANGES_REQUESTED** [review]
- Nav to create page for DO instance, add more than one pool, unselected pools show validation warning (red exclamation next to pool name in the tabs). 
- In `master`, it's impossible to invalidate the page (disable create button) due to machine pool config, even with absolute junk (remove all roles from all pools, deleting all pools, removing required fields, etc). Will create an issue

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 3]
```diff
@@ -696,9 +698,19 @@ export default {
...
       }
 
+      if (this.customCredentialComponentRequired === false) {
+        return false;
+      }
+
       return true;
     },
 
+    // Only for extensions - extension can register a 'false' cloud credential to indicate that a cloud credential is not needed
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:708]: Done

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 4]
```diff
@@ -1193,6 +1214,7 @@ export default {
         remove: false,
         create: true,
         update: false,
+        uid:    name,
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:1217]: Will change to id

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 5]
```diff
@@ -2132,6 +2169,7 @@ export default {
               :name="obj.id"
               :label="obj.pool.name || '(Not Named)'"
               :show-header="false"
+              :error="!machinePoolValidation[obj.uid]"
```
[the remark this answers is not in the document]
Comment (nwmac) [reply shell/edit/provisioning.cattle.io.cluster/rke2.vue:2172]: Fixed by defaulting new machine pools to be valid

File: shell/edit/provisioning.cattle.io.cluster/MachinePool.vue [hunk 6]
```diff
@@ -117,8 +123,11 @@ export default {
 
     isWindows() {
       return this.value?.config?.os === 'windows';
-    }
+    },
 
+    modeWithBusy() {
```
Comment (richard-cox) [inline shell/edit/provisioning.cattle.io.cluster/MachinePool.vue:128]: Not used anymore

**Review by richard-cox — CHANGES_REQUESTED** [review]
Tested DO (creating, viewing, editing, etc) and with a few workarounds for the merge issues (?) everything worked fine. Visually checked out the other changes and they look fine

**Review by richard-cox — APPROVED** [review]

**Review by richard-cox — APPROVED** [review]

**Pull request merged**